### PR TITLE
PYTHON-3295 Add CSOT docs page

### PR DESCRIPTION
**Pull request by ShaneHarvey**

### CODE REVIEW

File: doc/examples/timeouts.rst [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+The :meth:`~pymongo.timeout` applies to all pymongo operations within the block.
+The following example ensures that both the `insert` and the `find` complete
+within 10 seconds total, or raise a timeout error::
+
+  with pymongo.timeout(10):
+      coll.insert_one({"name": "Nunu"})
+      coll.find_one({"name": "Nunu"})
+
+When nesting :func:`~pymongo.timeout`, the newly computed deadline is capped to at most
```
Comment (blink1073) [inline doc/examples/timeouts.rst:33]: This was hard to parse, perhaps: "the nested deadline is still capped by the outer deadline".

File: doc/examples/timeouts.rst [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+      await coll.find_one({"name": "Nunu"})
+
+Troubleshooting
+---------------
+
+There are many timeout errors that can be raised depending on when the timeout
+expires. In code, these can be identified with the :attr:`pymongo.errors.PyMongoError.timeout`
+property. Some specific timeout errors examples are described below.
+
+The following error means that the client was unable to find an available server
```
Comment (blink1073) [inline doc/examples/timeouts.rst:133]: I would replace "The following error means that" with "When" for better flow in all of these examples

File: doc/examples/timeouts.rst (hunk 1 above)
Comment (ShaneHarvey) [reply doc/examples/timeouts.rst]: Done.

File: doc/examples/timeouts.rst (hunk 2 above)
Comment (ShaneHarvey) [reply doc/examples/timeouts.rst]: Fixed.

**Review by ShaneHarvey — COMMENTED** [review]
Ready for another look.

**Review by blink1073 — APPROVED** [review]
LGTM!
